src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1259,6 +1263,16 @@ bool layout_optimizer::are_layouts_suitable_for_onednn(program_node& node) {
     return true;
 }
 
+bool layout_optimizer::is_primitive_implemented_for_onednn(program_node& node) {
+    if (node.is_type<fully_connected>() || node.is_type<gemm>() || node.is_type<pooling>() ||
```

Comment:
How about add this querying function in primitive_type_base.h : 
as `does_possible_implementation_exist(const cldnn::program_node& node, const kernel_impl_params& impl_param, impl_types impl_type) ?`

Reply:
Sounds good. Shall we follow-up on the next PR @byungilm ?